src/edu/csus/ecs/pc2/clics/API202306/ContestService.java

```diff
@@ -420,11 +421,22 @@ private Response HandleContestThawTime(SecurityContext sc, String contestId, Str
         // thaw time present, validate now
         GregorianCalendar thawTime = getDate(contestId, thawTimeValue);
         if (thawTime != null) {
-            // Set thaw time to this time.
-            // TODO: tell PC2 to thaw the contest at the given time.
-            controller.getLog().log(Log.WARNING, LOG_PREFIX + contestId + ": setting of contest thaw time is not implemented");
-            return Response.status(Status.NOT_MODIFIED).entity("Unable to set contest thaw time to " + thawTime.toString()).build();
+            Date thawDate = thawTime.getTime();
+            String thawStr = Utilities.getIso8601formatterWithMS().format(thawDate);
+            // get the local model's ContestInformation sincd we are modifying the thaw time
+            ContestInformation ci = model.getContestInformation();
+            if (ci != null) {
+                // set the new start date/time into the ContestInformation
+                ci.setThawed(thawDate);
```

Comment:
The preceding comment says we are setting the "new start date/time", but I don't think that's what's actually being set here.  (Copy/paste without an update?)